src/main/java/com/flightstats/hub/spoke/FileSpokeStore.java

```diff
@@ -262,20 +283,22 @@ public void getNext(String channel, String startKey, int count, OutputStream out
             String minuteUrl = minutePath.toUrl();
             String minute = channelPath + minuteUrl;
             log.trace("minute {}", minute);
-            String[] items = new File(minute).list();
-            if (items != null) {
-                for (String item : items) {
-                    String keyFromPath = spokeKeyFromPath(minuteUrl + "/" + item);
-                    if (firstMinute) {
-                        ContentKey key = ContentKey.fromUrl(keyFromPath).get();
-                        if (key.compareTo(start) > 0) {
-                            found++;
-                            writeKey(output, channel + "/" + keyFromPath);
-                        }
-                    } else {
+            String[] items = Arrays.stream(
+                            Optional.ofNullable(new File(minute).list())
+                                    .orElse(new String[]{}))
+                    .filter(this::doesNotEndWith)
+                    .toArray(String[]::new);
+            for (String item : items) {
```

Comment:
Another case where I think you can use `foreach` on the stream. Unless there's a reason to do it in the for loop that I'm not seeing? I'm not seeing anything modified within in the for loop that would prevent being in a lambda.